### PR TITLE
#63 Extracted common configurations to base config.

### PR DESCRIPTION
**Pull request by Monteth**
I created base config file containing common rules, plugins and other options that are used in both default and typescript config. 

Changes were requested by @radekmie   in issue #63.

This repository doesn't specify PR template for structural changes, so if I did something wrong or if I should provide more info  - please let me know ;)

### CODE REVIEW

File: configs/default.yaml [hunk 1]
```diff
@@ -1,80 +1,19 @@
...
-  - eslint:recommended
   - prettier
-  - prettier/react
+  - ./base
 
 plugins:
   - babel
-  - prettier
-  - react
   - vazco
```
Comment (MacRusher) [inline configs/default.yaml:11]: `vazco` plugin is in the `base` config, so you should be able to remove it from here

File: configs/default.yaml [hunk 2]
```diff
@@ -1,80 +1,19 @@
...
-  react/no-access-state-in-setstate: 2
-  react/no-this-in-sfc: 2
-  react/react-in-jsx-scope: 2
-  react/self-closing-comp: 2
-  react/sort-comp: 2
-  strict: [0, "global"]
-  valid-jsdoc: 1
-  vazco/no-console: [1, {disallow: ["info", "time", "timeEnd"]}]
-  yoda: [1, "never"]
+      trailingComma: none
```
Comment (MacRusher) [inline configs/default.yaml]: Please add a newline at the end of the file. There is no eslint here to fix this automatically for you ;-)

Comment (MacRusher) [inline configs/default.yaml]: The prettier rules are different for `default` and `typescript`. I think we should unify them as well.

As much as I'm a fan of `trailingComma: all`, the `none` setting was in the original config, so it should be applied in both cases.

Comment (Monteth) [reply configs/default.yaml]: In principle, I agree with you, but in this PR I don't want to change any rule. We can discuss this in dedicated PR.

Comment (radekmie) [reply configs/default.yaml]: I'm 99% sure this already changes something - maybe we could use this opportunity to unify them even more?

Comment (Monteth) [reply configs/default.yaml]: I checked it out and "Add TypeScript config" was accepted with 0 thumbs up (and probably still legally xD), so I'm totally OK with changing it here, even without any special ceremony.

**Review by radekmie — APPROVED** [review]
I'm just thinking... Would it make sense to make `typescript` config extend the `default` one? Are there any problems with that?

**Comment by Monteth** [comment]
> I'm just thinking... Would it make sense to make `typescript` config extend the `default` one? Are there any problems with that?

`default` config uses `babel` eslint plugin and I don't think it would be profitable to attach it to typescript config.

**Comment by radekmie** [comment]
As I see, we're not using any [`eslint-plugin-babel`](https://github.com/babel/eslint-plugin-babel) rules, just the [`babel-eslint`](https://github.com/babel/babel-eslint) parser (which is actually being migrated to the Babel monorepo). TypeScript config overrides the parser as well, so it shouldn't use the `babel-eslint` one at all. @Monteth could you confirm that we are not using `eslint-plugin-babel` at all? If so, we could make `typescript` extend `default`.

**Comment by Monteth** [comment]
All of `eslint-plugin-babel` rules have `babel/` prefix and we don't have any such a rule. Also, running eslint without `babel` plugin does not generate any errors.

**Comment by Monteth** [comment]
OK so I moved base rules to `default.yaml`, changed `typescript.yaml` extends base to 'default',  removed `typescript` prettier rule override ( effectively set `trailingComma: none` - as it was set in parent config). I also added eslint settings to detect react version - it was set in `typescript` config and I moved it to `default` to fulfill warning message showing up while running eslint with `default` settings.

@MacRusher @radekmie

**Review by MacRusher — APPROVED** [review]

**Comment by Monteth** [comment]
We figured out that  previous "extends" config in `typescript.yaml` did't overwrite default eslint config. Consequently some of duplicated rules have been pointed out twice, sometimes with different level. We decided to overwrite them by `eslint-recommended` config, but this solution has its drawbacks - manually defined rules in `default.yaml` are overwritten by it. To fix this we have to check manually if changes introduced by this action are acceptable, if not - add another overwrite on the top level.

**Review by radekmie — APPROVED** [review]

**Pull request merged**